src/twisted/conch/test/test_filetransfer.py

```diff
@@ -572,6 +576,46 @@ def test_openDirectoryIteratorDeprecated(self):
         self.assertEqual(message, warnings[0]['message'])
 
 
+    @defer.inlineCallbacks
+    def test_closedConnectionCancelsRequests(self):
+        """
+        If there are requests outstanding when the connection
+        is closed for any reason, they should fail.
+        """
+
+        d = self.client.openFile(b"testfile1", filetransfer.FXF_READ, {})
+        self._emptyBuffers()
+        fh = yield d
+
+        # Intercept the handling of the read request on the server side
+        gotReadRequest = []
+
+        def _slowRead(offset, length):
+            self.assertEqual(gotReadRequest, [])
+            d = defer.Deferred()
+            gotReadRequest.append(d)
```

Comment:
this is ok... but what I am doing in similar cases is just use something like `gotReadRequest.append(True)` 

With your implementation, when I read the code I was thinking that you are doing something with the new Deferred as placed in the list.

Reply:
Yes, I'm not sure why I did that. I copied that piece of test code from elsewhere in Twisted. I'll clean it up.